### ISSUE TITLE
Option to alert when device goes out of range

### ISSUE THREAD

**Issue by cpfeiffer** [Freeyourgadget/Gadgetbridge#38]
It would be nice to have an option to get a notification *before* the connection between mobile and device (e.g. Mi Band) breaks.

This could be useful to detect that the mobile has been forgotten somewhere or stolen.

**Comment by danielegobbetti** [comment]
Related to this, what could perhaps help in case of a temporary non reachability of the band is to

- try to reconnect when a notification has been received by gadgetbridge

**Comment by cpfeiffer** [comment]
We actually have that already. The lowlevel bluetooth implementation already supports automatic reconnection (and we use that). I have no idea how long it attempts to do that or if it ever gives up doing that. 

Nor how much battery that consumes, but I haven't had many problems with this.

Additionally we support automatic connection whenever a notification is received.

**Comment by danielegobbetti** [comment]
Oh i had no idea. But it never happened to me to experience a reconnection when receiving a message/call/whatever

About battery life I don't see an impact as well, and perhaps it doesn't work on my galaxy nexus since BTLE support was done by the community (or perhaps it has too little memory to handle the background self-reconnection). I'll take a closer look to gadgetbridge.log file!


Il 22 maggio 2015 23:28:00 CEST, Carsten Pfeiffer <notifications@github.com> ha scritto:
>We actually have that already. The lowlevel bluetooth implementation
>already supports automatic reconnection (and we use that). I have no
>idea how long it attempts to do that or if it ever gives up doing that.
>
>
>Nor how much battery that consumes, but I haven't had many problems
>with this.
>
>Additionally we support automatic connection whenever a notification is
>received.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/Freeyourgadget/Gadgetbridge/issues/38#issuecomment-104778702

--

**Comment by cpfeiffer** [comment]
Yes, please have  a look at the log file. Note that I disabled logging to file by default, yesterday. You have to explicitly enable it in the settings, now.

**Comment by tcs-ulli** [comment]
reconnect dont works for me too, at work iam often not in range of my phone

**Comment by cpfeiffer** [comment]
@tcs-ulli Do you mean automatic reconnect when coming in reach again or do you mean when you're in reach and a notification comes in?

When a notification arrives, a connection attempt is always made.

Without a notification it depends on the lowlevel bluetooth stack how long and how often a reconnection occurs.

**Comment by ashimokawa** [comment]
@cpfeiffer 
What you write is true for the Mi Band but totally different for the Pebble. There is no connection attempt made, when a notification comes in. And also the low level bluetooth stack does not play a role here.

@tcs-ulli 
Which device are you using?

**Comment by tcs-ulli** [comment]
When i leave my desk at work, go out of range, and come back (in range) gadgetbridge never reconnect to my pebble. Also when a message arrives not. 
I use a GT-I9300 with Cyanogen Android 4.4.4 and Gadgetbridge 0.4.3 from fDroid

**Comment by tcs-ulli** [comment]
Sry for the late answer and nice work :)

**Comment by cpfeiffer** [comment]
@ashimokawa the reconnection upon received notification should be trivial to add for the pebble:
PebbleSupport:
```
@Override
public boolean useAutoConnect() {
    return true;
}
```

**Comment by cutec-chris** [comment]
Hi,
What about this ? 
Sounds good :)

**Comment by cpfeiffer** [comment]
@cutec-chris Do you mean the actual issue (alert when device out of range) or do you mean auto-connect with the pebble?

**Comment by cutec-chris** [comment]
autoconnect with pebble

**Comment by tcs-ulli** [comment]
What about cpfeifers sollution can it included in one of the next releases ?

**Comment by ashimokawa** [comment]
@tcs-ulli 
It does not work yet, not as simple as @cpfeiffer suggested.
